http_comms/comms.go

```diff
@@ -456,7 +456,7 @@ func (self *HTTPConnector) ReKeyNextServer(ctx context.Context) {
 			if err == nil {
 				return
 			}
-
+			self.logger.Info("While rekeying %v: %v", self.urls[self.current_url_idx], err)
```

Comment:
You can't access this without a lock here. You should add that to the place where the error is generated and just log the error here.